cpu/esp32/periph/adc.c

```diff
@@ -271,11 +270,11 @@ static void _adc_module_init(void)
 
 void adc_print_config(void)
 {
-    ets_printf("\tADC\t\tpins=[ ");
+    printf("\tADC\t\tpins=[ ");
```

Comment:
Why not use the following ?
```suggestion
    DEBUG("\tADC\t\tpins=[ ");
```

Here and below ? debug.h is included in this file so that would only be displayed when ENABLE_DEBUG is set. Maybe I'm missing something but that seems strange at first sight.

Reply:
The reason why `printf` is used instead of `DEBUG` is that https://github.com/RIOT-OS/RIOT/blob/178d0aace20784e69a641c546c845fc8dce0fae1/boards/common/esp32/board_common.c#L57 prints the whole configuration for all ESP32 boards, for example
```
	ADC		pins=[ 0 2 4 12 13 14 15 25 26 27 32 33 34 35 36 39 ]
	PWM_DEV(0)	channels=[ 0 2 4 16 17 ]
	PWM_DEV(1)	channels=[ 27 32 33 ]
	I2C_DEV(0)	scl=22 sda=21
	SPI_DEV(0)	VSPI sck=18 miso=19 mosi=23 cs=5
	SPI_DEV(1)	HSPI sck=14 miso=12 mosi=13 cs=15
	UART_DEV(0)	txd=1 rxd=3
	UART_DEV(1)	txd=10 rxd=9
	LED		pins=[ ]
	BUTTONS		pins=[ 0 ]
```
which is quite useful to know which pin is what and how it can be used.

For that purpose each peripheral driver in `cpu/esp32/periph` provides a function `*_print_config` which is called if the according `periph_*` module is used. Because the board configuration is always printed and not only if debugging for the according file is enabled, `printf` is used here.

It is the same for ESp8266 and until now nobody has complained about that.

Reply:
It's only an explanation and no justification. If you think it's nonsense, we should discuss it. I could live with an approach where `print_board_config` is only called when `DEVELHELP` is enabled. But in that case, `printf` would be also the right one.

Reply:
BTW, I wished I had such a `print_board_config` for other boards too.

Reply:
Ok, in the meantime I checked the code to know from where it was called and got to the same conclusion.
If we want to limit the firmware size, I think this should be enabled just for debugging purpose and not all the time. On esp32, size is not really a problem.

Reply:
> I think this should be enabled just for debugging purpose and not all the time. On esp32, size is not really a problem.

Is it an issue or not :wink: I could add the following in https://github.com/RIOT-OS/RIOT/blob/178d0aace20784e69a641c546c845fc8dce0fae1/cpu/esp32/startup.c#L330
```c
if DEVELHELP
    print_board_config();
endif
```
and all `*_print_config` functions are optimized out if `DEVELHELP` is not defined.

Reply:
Yes what you propose would basically be the idea.

Reply:
Maybe it is better to use `NDEBUG` instead of `DEVELHELP`. `DEVELHELP` has to be enabled explicitly. IMO, the board config should be printed by default during the development. The optimized production version would be made with `NDEBUG` defined. This would be the same approach as with `assert`.